.github/workflows/pull-requests.yml

```diff
@@ -0,0 +1,96 @@
+name: pull-requests
+
+on:
+    pull_request:
+
+jobs:
+    style:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: Install Node.js
+              uses: actions/setup-node@v4
+              with:
+                  node-version: 20
+                  cache: yarn
+
+            - name: "Install Foundry"
+              uses: "foundry-rs/foundry-toolchain@v1"
+
+            - name: Install dependencies
+              run: yarn
+
+            - name: Get changed files
+              id: changed-files
+              uses: tj-actions/changed-files@v44
+              with:
+                  files_yaml: |
+                      contracts:
+                        - packages/contracts/**/*.{js,json,ts,sol}
+                      docs:
+                        - apps/docs/**/*
+                      to_format:
+                        - '**/*.{cjs,js,json,jsx,md,mdx,sol,ts,tsx,yaml,yml}'
+                      to_lint:
+                        - '**/*.{cjs,js,jsx,ts,tsx}'
+
+            - if: steps.changed-files.outputs.contracts_any_changed == 'true'
+              name: Compile and lint contracts
+              run: |
+                  yarn compile:contracts
+                  yarn workspace excubiae-contracts lint
+
+            - if: steps.changed-files.outputs.docs_any_changed == 'true'
+              name: Build and format docs
+              run: |
+                  yarn workspace excubiae-docs build
+                  yarn workspace excubiae-docs format
+
+            - if: steps.changed-files.outputs.libraries_any_changed == 'true'
```

Comment:
`libraries` is defined in the other job and used here. Not sure if this works.

Reply:
good catch!